File: CLASE V - Listas Void/listas_int/c/nodo.c
#include <stdio.h>
#include <stdlib.h>
#include "nodo.h"


struct Nodo {

    int dato;
    struct Nodo * siguiente;

};


int getDato(NodoPtr nodo){

    return nodo->dato;
};

NodoPtr getSiguiente(NodoPtr nodo){

    return nodo->siguiente;
};

void setDato(NodoPtr nodo, int nuevoDato){

    nodo->dato = nuevoDato;
};

void setSiguiente(NodoPtr nodo, NodoPtr nuevoSiguiente){

    nodo->siguiente = nuevoSiguiente;

};

NodoPtr crearNodo(int dato, NodoPtr siguiente){


    NodoPtr nodo = (NodoPtr) malloc(sizeof(struct Nodo));

    nodo->dato = dato;
    nodo->siguiente = siguiente;

    return nodo;

};

void mostrarNodo(NodoPtr nodo){

    printf("{ %d }", nodo->dato);

};


void liberarNodo(NodoPtr nodo){

    free(nodo);


};

File: CLASE V - Listas Void/listas_int/golang/nodo.go
package main

import "fmt"

type Nodo struct {
	dato      int
	siguiente *Nodo
}

func crearNodo(dato int, siguiente *Nodo) *Nodo {
	return &Nodo{dato: dato, siguiente: siguiente}
}

func getDato(n *Nodo) int {
	return n.dato
}

func getSiguiente(n *Nodo) *Nodo {
	return n.siguiente
}

func setDato(n *Nodo, nuevoDato int) {
	n.dato = nuevoDato
}

func setSiguiente(n *Nodo, nuevoSig *Nodo) {
	n.siguiente = nuevoSig
}

func mostrarNodo(n *Nodo) {
	fmt.Printf("{ %d }", n.dato)
}

